### PR TITLE
refactor(userconfig): move regexp to var

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

moves various regexp to vars

## Why this way

it's cleaner this way

### CODE REVIEW

File: internal/schemautil/userconfig/apiconvert/toapi.go [hunk 1]
```diff
@@ -17,6 +17,16 @@ type resourceDatable interface {
 	IsNewResource() bool
 }
 
+var (
+	// dotAnyDotNumberRegExp is a regular expression that matches a string that begins with a dot and ends with a dot
+	// and a number.
+	dotAnyDotNumberRegExp = regexp.MustCompile(`.+\.[0-9]$`)
+
+	// dotNumberEOLOrDotRegExp is a regular expression that matches a string that begins with a dot and ends with a
+	// number and an end of line or a dot.
```
Comment (byashimov) [inline internal/schemautil/userconfig/apiconvert/toapi.go:26]: Please rephrase.

File: internal/schemautil/userconfig/apiconvert/toapi.go [hunk 2]
```diff
@@ -17,6 +17,16 @@ type resourceDatable interface {
...
+	dotNumberEOLOrDotRegExp = regexp.MustCompile(`\.\d($|\.)`)
```
Comment (byashimov) [inline internal/schemautil/userconfig/apiconvert/toapi.go:27]: It matches "dot, number, dot" anywhere and "dot, number" at the end of the line. Is that expected?

File: internal/schemautil/userconfig/stateupgrader/v0/serviceintegration/service_integration.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig/stateupgrader/v0/dist"
 )
 
-const serviceIntegrationEndpointRegExp = "^[a-zA-Z0-9_-]*\\/{1}[a-zA-Z0-9_-]*$"
+var serviceIntegrationEndpointRegExp = regexp.MustCompile(`^[a-zA-Z0-9_-]*/[a-zA-Z0-9_-]*$`)
```
Comment (byashimov) [inline internal/schemautil/userconfig/stateupgrader/v0/serviceintegration/service_integration.go:15]: This matches even to a slash `/` is that expected?

**Review by byashimov — CHANGES_REQUESTED** [review]

File: internal/schemautil/userconfig/apiconvert/toapi.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/apiconvert/toapi.go:26]: I believe it is clear enough

File: internal/schemautil/userconfig/stateupgrader/v0/serviceintegration/service_integration.go (hunk 3 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/stateupgrader/v0/serviceintegration/service_integration.go:15]: it is

**Review by byashimov — APPROVED** [review]

**Comment by byashimov** [comment]
lgtm

**Pull request merged**